File: app-aws-lambda-example-go/services/AppExampleRestService.go
package services

import (
	"github.com/exproletariy/pip-services3-containers-examples/app-aws-lambda-example-go/logic"
	"net/http"

	crefer "github.com/pip-services3-go/pip-services3-commons-go/refer"
	rpc "github.com/pip-services3-go/pip-services3-rpc-go/services"
)

type AppExampleRestService struct {
	*rpc.RestService
	controller *logic.AppExampleController
}

func NewAppExampleRestService() *AppExampleRestService {
	c := AppExampleRestService{}
	c.RestService = rpc.InheritRestService(&c)
	c.BaseRoute = "/example"
	c.DependencyResolver.Put("controller", crefer.NewDescriptor("app-example", "controller", "*", "*", "1.0"))
	return &c
}

func (c *AppExampleRestService) SetReferences(references crefer.IReferences) {

	c.RestService.SetReferences(references)
	depRes, depErr := c.DependencyResolver.GetOneRequired("controller")
	if depErr == nil && depRes != nil {
		c.controller = depRes.(*logic.AppExampleController)
	}
}

func (c *AppExampleRestService) greeting(res http.ResponseWriter, req *http.Request) {
	name := req.URL.Query().Get("name")
	result, err := c.controller.Greeting(name)
	c.SendResult(res, req, result, err)

}

func (c *AppExampleRestService) Register() {
	c.RegisterRoute("get", "/greeting", nil, c.greeting)
}

File: app-process-container-example-nodex/src/services/AppExampleRestService.ts
import { RestService } from 'pip-services3-rpc-nodex';
import { AppExampleController } from '../logic';
import { ConfigParams, Descriptor, IReferences } from 'pip-services3-commons-nodex';

export class AppExampleRestService extends RestService{
    private _controller: AppExampleController;

    public constructor() {
        super();
        this._baseRoute = '/example';
        this._dependencyResolver.put('controller', new Descriptor('app-example', 'controller', 'default', '*', '*'));
    }

    public configure(config: ConfigParams): void {
        super.configure(config);
    }

    public setReferences(references: IReferences): void {
        super.setReferences(references);
        this._controller = this._dependencyResolver.getOneRequired<AppExampleController>('controller');
    }

    private greeting(req, res) {
        let promise = this._controller.greeting(req.query.name);
        this.sendResult(req, res, promise);
    }

    register(): void {
        this.registerRoute('get', '/greeting', null, this.greeting);
    }

}